servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpSerializers.java

```diff
@@ -0,0 +1,375 @@
+     *
+     * @param charset The character encoding to use for serialization.
+     * @return a {@link HttpStreamingSerializerDeserializer} that serializes {@link String}s with
+     * {@code charset} encoding using fixed {@code int} length delimited framing.
+     * @see VarIntLengthStreamingSerializer
+     */
+    public static HttpStreamingSerializerDeserializer<String> textSerializerVarLen(Charset charset) {
+        if (UTF_8.equals(charset)) {
+            return TEXT_STREAMING_VAR_LEN_UTF_8;
+        } else if (US_ASCII.equals(charset)) {
+            return TEXT_STREAMING_VAR_LEN_ASCII;
+        }
+        final int maxBytesPerChar = (int) charset.newEncoder().maxBytesPerChar();
+        CharSequence contentType = newAsciiString(APPLICATION_TEXT_VARINT + "; charset=" + charset.name());
+        return streamingSerializer(new VarIntLengthStreamingSerializer<>(stringSerializer(charset),
+                        str -> str.length() * maxBytesPerChar),
+                headers -> headers.set(CONTENT_TYPE, contentType),
+                headers -> hasContentType(headers, APPLICATION_TEXT_VARINT, charset));
+    }
+
+    /**
+     * Create a {@link HttpStreamingSerializer} that serializes {@link String}. This method is useful if the payload
+     * body is provided in {@link String} and the {@link HttpHeaderNames#CONTENT_TYPE} is known a-prior
+     * (e.g. streaming raw json data from a stream of {@link String}s). Deserialization should be done using
+     * the a-prior knowledge to use a compatible {@link HttpStreamingDeserializer}.
+     * @param charset The character encoding to use for serialization.
+     * @param headersSerializeConsumer Sets the headers to indicate the appropriate encoding and content type.
+     * @return a {@link HttpStreamingSerializer} that uses a {@link Serializer} for serialization.
+     */
+    public static HttpStreamingSerializer<String> stringStreamingSerializer(
+            Charset charset, Consumer<HttpHeaders> headersSerializeConsumer) {
+        final int maxBytesPerChar = (int) charset.newEncoder().maxBytesPerChar();
+        return streamingSerializer(stringSerializer(charset), str -> str.length() * maxBytesPerChar,
+                headersSerializeConsumer);
+    }
+
+    /**
+     * Create a {@link HttpStreamingSerializer} that serializes {@code byte[]}. This method is useful if the payload
+     * body is provided in {@code byte[]} and the {@link HttpHeaderNames#CONTENT_TYPE} is known a-prior
+     * (e.g. streaming raw json data from a stream of {@code byte[]}s). Deserialization should be done using
+     * the a-prior knowledge to use a compatible {@link HttpStreamingDeserializer}.
+     * @param headersSerializeConsumer Sets the headers to indicate the appropriate encoding and content type.
+     * @return a {@link HttpStreamingSerializer} that uses a {@link Serializer} for serialization.
+     */
+    public static HttpStreamingSerializer<byte[]> bytesStreamingSerializer(
```

Comment:
Can we also support deserializer?

Reply:
we can as a followup. we don't expose `byte[]` so I'm not sure its necessary. this method is to accommodate "stream content with a-prior known framing in pre-existing data type" and the other side is expected to use the corresponding deserialization directly (e.g. json/xml/etc.).